src/Product/SearchProvider.php

```diff
@@ -52,14 +52,21 @@ class SearchProvider implements FacetsRendererInterface, ProductSearchProviderIn
      */
     private $facetsSerializer;
 
+    /**
+     * @var SearchFactory
+     */
+    private $searchFactory;
+
     public function __construct(
         Ps_Facetedsearch $module,
         Filters\Converter $converter,
-        URLSerializer $serializer
+        URLSerializer $serializer,
+        ?SearchFactory $searchFactory = null
```

Comment:
```suggestion
        SearchFactory $searchFactory = null
```

This module is still compatible with older Ps version